### PR TITLE
Refactor TesteeEntityTest and TesteesMapperTest

### PR DESCRIPTION
**Pull request by Yarik2308**
https://github.com/malinink/easy-tests/issues/228

- [x] TesteeEntityTest
- [x] TesteesMapperTest
- [x] Some AssertEquals and AssertNotEquals

### CODE REVIEW

**Comment by malinink** [comment]
@Yarik2308 Add tasks list into first comment.
All tasks accomplishment give me a reason to look into PR.

**Comment by Yarik2308** [comment]
@malinink The task list is in the issue.

**Comment by malinink** [comment]
@Yarik2308 Task list with checkbox must be in PR

File: src/test/java/easytests/support/SubjectsSupport.java [hunk 1]
```diff
@@ -68,7 +68,6 @@ private SubjectEntity getEntityMock(Object[] data) {
                 (Integer) data[3]
         );
     }
-
```
Comment (malinink) [inline src/test/java/easytests/support/SubjectsSupport.java]: revert SubjectSupport edition.

File: src/test/java/easytests/support/TesteesSupport.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        Mockito.when(testeeModel.getId()).thenReturn(id);
+        Mockito.when(testeeModel.getFirstName()).thenReturn(firstName);
+        Mockito.when(testeeModel.getLastName()).thenReturn(lastName);
+        Mockito.when(testeeModel.getSurname()).thenReturn(surName);
+        Mockito.when(testeeModel.getGroupNumber()).thenReturn(groupNumber);
+        Mockito.when(testeeModel.getQuiz()).thenReturn(new QuizModelEmpty(quizId));
+        return testeeModel;
+    }
+
+    public void assertEquals(TesteeEntity expected, TesteeEntity actual) {assertEquals(expected, actual, false);}
```
Comment (malinink) [inline src/test/java/easytests/support/TesteesSupport.java]: make it multiline

File: src/test/java/easytests/support/TesteesSupport.java [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+    public void assertEqualsWithoutId(TesteeEntity expected, TesteeEntity actual) {assertEquals(expected, actual, true);}
```
Comment (malinink) [inline src/test/java/easytests/support/TesteesSupport.java]: multiline

File: src/test/java/easytests/core/mappers/TesteesMapperTest.java [hunk 4]
```diff
@@ -1,116 +1,96 @@
...
  */
 public class TesteesMapperTest extends AbstractMapperTest {
 
+    protected TesteesSupport testeesSupport = new TesteesSupport();
+
     @Autowired
     private TesteesMapper testeesMapper;
 
     @Test
+    // 20min
```
Comment (malinink) [inline src/test/java/easytests/core/mappers/TesteesMapperTest.java]: no no, you should not add time remarks in code, in PR please!

File: src/test/java/easytests/core/mappers/TesteesMapperTest.java [hunk 5]
```diff
@@ -1,116 +1,96 @@
 package easytests.core.mappers;
 
 import easytests.core.entities.TesteeEntity;
+import easytests.support.TesteesSupport;
+
```
Comment (malinink) [inline src/test/java/easytests/core/mappers/TesteesMapperTest.java]: correct import order, remove empty lines.

File: src/test/java/easytests/core/entities/TesteeEntityTest.java [hunk 6]
```diff
@@ -1,50 +1,29 @@
...
-import org.meanbean.test.HashCodeMethodTester;
-import org.mockito.Mockito;
 
 /**
  * @author DoZor-80
  */
-public class TesteeEntityTest {
+public class TesteeEntityTest extends AbstractEntityTest{
+
+    protected TesteesSupport testeesSupport = new TesteesSupport();
```
Comment (malinink) [inline src/test/java/easytests/core/entities/TesteeEntityTest.java]: add empty line after statement

File: src/test/java/easytests/core/mappers/TesteesMapperTest.java [hunk 7]
```diff
@@ -1,116 +1,96 @@
...
 
     @Test
+    // 20min
     public void testFindAll() throws Exception {
         final List<TesteeEntity> testeesEntities = this.testeesMapper.findAll();
         Assert.assertEquals(3, testeesEntities.size());
+
+        Integer index = 0;
+        for (TesteeEntity testeeEntity: testeesEntities){
+            final TesteeEntity testeeFixtureEntity = this.testeesSupport.getEntityFixtureMock(index);
```
Comment (malinink) [inline src/test/java/easytests/core/mappers/TesteesMapperTest.java]: new naming

File: src/test/java/easytests/core/mappers/TesteesMapperTest.java [hunk 8]
```diff
@@ -1,116 +1,96 @@
...
 
     @Test
     public void testFind() throws Exception {
         final TesteeEntity testee = this.testeesMapper.find(1);
-        Assert.assertEquals((long) 1, (long) testee.getId());
-        Assert.assertEquals("FirstName1", testee.getFirstName());
-        Assert.assertEquals("LastName1", testee.getLastName());
-        Assert.assertEquals("Surname1", testee.getSurname());
-        Assert.assertEquals((long) 1, (long) testee.getQuizId());
+        final TesteeEntity testeeFixtureEntity = this.testeesSupport.getEntityFixtureMock(0);
```
Comment (malinink) [inline src/test/java/easytests/core/mappers/TesteesMapperTest.java]: new naming

File: src/test/java/easytests/core/mappers/TesteesMapperTest.java [hunk 9]
```diff
@@ -1,116 +1,96 @@
...
+
+        this.testeesSupport.assertEquals(testeeFixtureEntity, testee);
     }
 
     @Test
+    //50min
     public void testFindByQuizId() throws Exception {
```
Comment (malinink) [inline src/test/java/easytests/core/mappers/TesteesMapperTest.java]: look into SubjectMapper FindByUserId, fix.

File: src/test/java/easytests/core/mappers/TesteesMapperTest.java [hunk 10]
```diff
@@ -1,116 +1,96 @@
...
     @Test
+    //1hr
     public void testInsert() throws Exception{
         final ArgumentCaptor<Integer> id = ArgumentCaptor.forClass(Integer.class);
-        final String firstName = "FirstName";
-        final String lastName = "LastName";
-        final String surname = "Surname";
-        final Integer groupNumber = 307;
-        final Integer quizId = 19;
+        final TesteeEntity testeeAdditionalEntity = this.testeesSupport.getEntityAdditionalMock(0);
```
Comment (malinink) [inline src/test/java/easytests/core/mappers/TesteesMapperTest.java]: namings

File: src/test/java/easytests/core/mappers/TesteesMapperTest.java [hunk 11]
```diff
@@ -1,116 +1,96 @@
...
-        final Integer id = 1;
-        final String firstName = "NewFirstName";
-        final String lastName = "NewLastName";
-        final String surname = "NewSurname";
-        final Integer groupNumber = 308;
-        final Integer quizId = 19;
-
-        TesteeEntity testeeEntity = this.testeesMapper.find(id);
+        final TesteeEntity testeeAdditionalEntity = this.testeesSupport.getEntityAdditionalMock(1);
+        final Integer id = testeeAdditionalEntity.getId();
```
Comment (malinink) [inline src/test/java/easytests/core/mappers/TesteesMapperTest.java]: We could avoid `id` use `testeeAdditionalEntity.getId()` instead

File: src/test/java/easytests/core/mappers/TesteesMapperTest.java [hunk 12]
```diff
@@ -1,116 +1,96 @@
...
-        Assert.assertEquals(lastName, testeeEntity.getLastName());
-        Assert.assertEquals(surname, testeeEntity.getSurname());
-        Assert.assertEquals(groupNumber,testeeEntity.getGroupNumber());
-        Assert.assertEquals(quizId, testeeEntity.getQuizId());
+        this.testeesSupport.assertNotEqualsWithoutId(testeeAdditionalEntity, testeeEntity);
+
+        this.testeesMapper.update(testeeAdditionalEntity);
+        final TesteeEntity updatedTesteeEntity = this.testeesMapper.find(id);
+
+        this.testeesSupport.assertEquals(testeeAdditionalEntity, updatedTesteeEntity);
```
Comment (malinink) [inline src/test/java/easytests/core/mappers/TesteesMapperTest.java]: Fix delete to use fixtures also

File: src/test/java/easytests/core/mappers/TesteesMapperTest.java (hunk 11 above)
Comment (janchk) [reply src/test/java/easytests/core/mappers/TesteesMapperTest.java]: Don't quite understand.

File: src/test/java/easytests/core/mappers/TesteesMapperTest.java (hunk 9 above)
Comment (janchk) [reply src/test/java/easytests/core/mappers/TesteesMapperTest.java]: What exactly do I need to fix?

Comment (malinink) [reply src/test/java/easytests/core/mappers/TesteesMapperTest.java]: create list by adding concrete fixtures, not in loop.
get list from find function.
assert equals two lists.

File: src/test/java/easytests/core/mappers/TesteesMapperTest.java (hunk 11 above)
Comment (malinink) [reply src/test/java/easytests/core/mappers/TesteesMapperTest.java]: @janchk  ride of id variable here, just use testeeAdditionalEntity.getId() instead

File: src/test/java/easytests/core/mappers/TesteesMapperTest.java (hunk 9 above)
Comment (janchk) [reply src/test/java/easytests/core/mappers/TesteesMapperTest.java]: @malinink Константин Александрович. Уже 5ый час ночи и у меня полыхнуло. Вы пишите "get list from find function.", но это невозможно по причине того, что "TesteeEntity findByQuizId(Integer quizId);" возвращает именно сущность, а не массив сущностей как это делает List<SubjectEntity> findByUserId(Integer userId). Кроме того, мой метод работал логично и за каким лешим его потребовалось менять, мне не понятно.

**Comment by Yarik2308** [comment]
@malinink seems like everything is done.

**Comment by malinink** [comment]
@Yarik2308 @janchk pls look at my cs changes.

**Pull request merged**